--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2534,6 +_,20 @@
     }
 
     @Override
+    public @Nullable Entity getEntity(final Integer entityId) {
+        Preconditions.checkArgument(entityId != null, "entityId cannot be null");
+
+        for (ServerLevel world : this.getServer().getAllLevels()) {
+            net.minecraft.world.entity.Entity entity = world.getEntity(entityId);
+            if (entity != null) {
+                return entity.getBukkitEntity();
+            }
+        }
+
+        return null;
+    }
+
+    @Override
     public org.bukkit.advancement.Advancement getAdvancement(NamespacedKey key) {
         Preconditions.checkArgument(key != null, "key cannot be null");
 
