--- a/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
@@ -26,6 +_,13 @@
         this.yHeadRot = buffer.readByte();
     }
 
+    // Octopus start - add constructor with entity id
+    public ClientboundRotateHeadPacket(int entityId, byte yHeadRot) {
+        this.entityId = entityId;
+        this.yHeadRot = yHeadRot;
+    }
+    // Octopus end - add constructor with entity id
+
     private void write(FriendlyByteBuf buffer) {
         buffer.writeVarInt(this.entityId);
         buffer.writeByte(this.yHeadRot);
