--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2425,6 +_,20 @@
         return server.getEntity(uuid);
     }
 
+    // Octopus start - get entity accessor by id
+
+    /**
+     * Retrieves an entity based on the provided entity ID.
+     *
+     * @param entityId the unique identifier of the entity to retrieve; must not be null
+     * @return the Entity corresponding to the specified ID, or null if no such entity exists
+     */
+    @Nullable
+    public static Entity getEntity(@NotNull Integer entityId) {
+        return server.getEntity(entityId);
+    }
+    // Octopus end - get entity accessor by id
+
     // Paper start
     /**
      * Gets the current server TPS
